cmd/csi-attacher/main.go

```diff
@@ -134,8 +135,17 @@ func main() {
 		os.Exit(1)
 	}
 	klog.V(2).Infof("CSI driver name: %q", csiAttacher)
-	metricsManager.SetDriverName(csiAttacher)
-	metricsManager.StartMetricsEndpoint(*metricsAddress, *metricsPath)
+
+	// Prepare http endpoint for metrics + leader election healthz
+	mux := http.NewServeMux()
+	cmm := metrics.NewCSIMetricsManagerForSidecar(csiAttacher)
+	cmm.RegisterToServer(mux, *metricsPath)
+	go func() {
```

Comment:
I'm not sure what the behavior would be to start the server before registering the leader election handler. Might be safer to do it after?

Reply:
I checked that both paths (metrics + leader check) work. I don't want to move the code further, because there would be two places to start the http server, in `then` and `else` branches of this `if`:

https://github.com/kubernetes-csi/external-attacher/blob/6b94e28d9479e4ebed65b3a373149ec054c5cac7/cmd/csi-attacher/main.go#L196-L205